aten/src/ATen/gen.py

```diff
@@ -360,6 +403,28 @@ def filter_by_extension(files, *extensions):
     return filtered_files
 
 
+def generate_per_op_registration(per_op_registrations):
+    if not options.per_op_registration:
+        return
+
+    # Ensure all whitelisted operators have a corresponding registration file.
+    # Generate an empty placeholder file for nonexistent operators, which might
+    # be registered manually instead of via codegen.
+    for whitelisted_op in op_registration_whitelist:
+        if whitelisted_op not in per_op_registrations:
+            per_op_registrations[whitelisted_op] = []
+
+    for opname, function_registrations in per_op_registrations.items():
+        fname = 'pt_op_register_%s.cpp' % opname.replace(':', '_')
+        # output_dependencies doesn't seem to be used, otherwise it is probably
+        # too late to declare output here?
```

Comment:
No it's not. The way that `output_dependencies` works is we do a dry run of what files we would write, but don't actually write any of them (that's what file manager manages: it will no-op if we're doing a dry run) so that the build system can account for it. So ideally, you'd have modified `declare_outputs` to have a new logic for computing what per op files you are going to generate, and then `generate_outputs` to actually do the generation. I'm not sure if you can conveniently do this with per-file op registration as I'm not sure if you have enough of the information in `declare_outputs` to say what files will be generated.

Reply:
Maybe, every file has to be specified in the whitelist command line? Then you can populate the set of written out files this way.